Add 'custom' key in entry-list for user-defined print functions

ToppDev opened the pull request:
This PR allows an additional field called `custom` in the entry-list which can be used in user-defined functions, e.g. `user-print-glossary`. For more information see the related issue #143.

I added some documentation for it in the README. Please feel free to suggest changes or make them directly in the PR branch.

Closes #143

quachpas commented:
Hello @ToppDev, thanks for the PR. 
It looks almost perfect to me. Can you implement `gls-custom` to get the attribute? Don't forge to export it in glossarium.typ as well.
About the CI, please run `tt update debug` that, or I can do it. I think you will have to do it on other tests too, like `style-entries`.

ToppDev commented:
I added the functions. Updating the test I will do in a bit.

However I ran into 2 small problem, which I want to discuss first.

1. All other `gls-...` functions have a `link` argument. I removed it in the `gls-custom` as it does not make sense. Or would you want to keep it? It has errors which need fixing then.

---

2. The `gls-custom` function returns a `content`, because of the `context` call. A user cannot access the values anymore if it is a dictionary before. So does the `gls-custom` function make sense at all?

```typst
entry-list: (
  (
    key: "c",
    short: $c$,
    description: (mkDesc("speed of light in vacuum", unit("meter per second"))),
    custom: (
      unit: unit("meter per second"),
      temp: "rsstr",
    ),
  ),
)
```

```typst
@c

#gls-custom("c")

Type: #type(gls-custom("c"))
```

renders to
![image](https://github.com/user-attachments/assets/fe897a60-bbf7-4e7b-8474-72b1245ff92b)

quachpas commented:
Hello, thanks for the quick update.

About 1., yes please keep the interface the same for everything function. I do not see a good reason to remove the ability to create a link, especially since custom can be anything, even a string, or simple content. It does not have to be a dictionary.

On 2., thank you! I just pushed 50385b1, which should provide you the necessary elements to make this PR even better with tests. I can suggest the following for a context test. Don't hesitate to mention that specifically in the docstring of gls-custom.
```typ
#import "../../themes/default.typ": *
#import "@preview/metro:0.3.0": *
#let entry-list = (
  (
    key: "c",
    short: $c$,
    description: unit("meter per second"),
    custom: (
      unit: unit("meter per second"),
      temp: "rsstr",
    ),
  ),
)

#register-glossary(entry-list)
#show: make-glossary

#gls-custom("c", ctx: false)

Type: #type(gls-custom("c", ctx: false))
```

I can eventually add a warning in the manual (docs are a work in progress) about disabling context wrapping if your custom value is already calling context.

You can rebase on master if possible. Otherwise I'll deal with the fallout.

ToppDev commented:
@quachpas thanks for your fast changes. I did the following now

- rebased on main
- changed the `gls-custom` (tried my best with the docstring)
-  added a unit test for the custom attribute
-  updated existing tests

hopefully all tests passing now. Could you review again?
